computer_vision/random_erasing_augmentation.py

```diff
@@ -0,0 +1,99 @@
+import glob
+import os
+import random
+from string import ascii_lowercase, digits
+import cv2
+
+
+# Params
+IMAGE_DIR = "1"
+OUTPUT_DIR = "out"
+MAX_RECTANGLE_WIDTH_PERCENT = 20  # (between 0 and 100)
+MAX_RECTANGLE_HEIGHT_PERCENT = 20  # (between 0 and 100)
+RECTANGLE_COLOR = "WHITE"  # ("BLACK" or "WHITE")
+
+
+def main() -> None:
+    """
+    Get images list from input dir.
+    Update new images.
+    Save images in output dir.
+    """
+    img_paths = get_dataset(IMAGE_DIR)
+    print("Processing...")
+    new_images, paths = update_images(
+        img_paths,
+        MAX_RECTANGLE_WIDTH_PERCENT,
+        MAX_RECTANGLE_HEIGHT_PERCENT,
+        RECTANGLE_COLOR,
+    )
+
+    for index, image in enumerate(new_images):
+        # Get random string code: '7b7ad245cdff75241935e4dd860f3bad'
+        letter_code = random_chars(32)
+        file_name = paths[index].split(os.sep)[-1].rsplit(".", 1)[0]
+        file_root = f"{OUTPUT_DIR}/{file_name}_NEW_{letter_code}"
+        cv2.imwrite(f"{file_root}.jpg", image, [cv2.IMWRITE_JPEG_QUALITY, 100])
+        print(f"Success {index+1}/{len(new_images)} with {file_name}")
+
+
+def get_dataset(img_dir: str) -> list:
+    """
+    - img_dir: Path to a folder that contains images
+    Return: List of images path
+    """
+    return glob.glob(os.path.join(img_dir, "*"))
+
+
+def update_images(
+    img_list: list,
```

Comment:
add this doctest  in `update_images` :-

    """
    Given a list of image paths, maximum rectangle width and height percentages,
    and a rectangle color, update each image by adding a random rectangle of the
    specified color and size. Returns a tuple containing the new images list and the
    corresponding path list.

    >>> img_list = ['1/image1.jpg', '1/image2.jpg', '1/image3.jpg']
    >>> maximum_rectangle_width_percent = 20
    >>> maximum_rectangle_height_percent = 20
    >>> rectangle_color = 'WHITE'
    >>> new_imgs_list, path_list = update_images(
    ...     img_list, maximum_rectangle_width_percent, maximum_rectangle_height_percent,
    ...     rectangle_color
    ... )
    >>> len(new_imgs_list)
    3
    >>> len(path_list)
    3
    >>> new_imgs_list[0].shape
    (360, 480, 3)
    >>> path_list[0]
    '1/image1.jpg'

    """

Reply:
I don't know how should I do these things (because other files in this directory don't contain them). Would you please add these yourself?

Reply:
yes.. as no images is there so we cant do in this way... kindly ask @cclauss to give suggestion.

Reply:
Please note that I implemented it using other files in this directory. If this one needs to change, other files need to change as well. I think we haven't reviewed the old pull requests of this directory.